model/channel.go

```diff
@@ -84,6 +84,19 @@ type DirectChannelForExport struct {
 	Members *[]string
 }
 
+// GetChannelsOptions contains options for getting channels
+type GetChannelsOptions struct {
+	// ChannelTypes optionally limits the types of channels to return.
+	// See model.CHANNEL*OPEN, model.CHANNEL_PRIVATE, model.CHANNEL_DIRECT, and model.CHANNEL*GROUP.
+	ChannelTypes []string
```

Comment:
If `ChannelTypes` is limited to this list, we should validate this. What do you think about adding a `IsValid() *AppError` method to `GetChannelsOptions`?